### ISSUE TITLE
`-Wpointer-bool-conversion` not reported in macro

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#60956]
```cpp
#define cb(...) cb_(__VA_ARGS__)
#define cb2(x) cb_(x)
void cb_(bool);

int main() {
  int i;
  cb_(&i); // warning
  cb(&i); // no warning
  cb2(&i);
}
```

```
<source>:6:8: warning: address of 'i' will always evaluate to 'true' [-Wpointer-bool-conversion]
  cb_(&i);
  ~~~  ^
```

https://godbolt.org/z/3GovP6r3n

**Comment by shafik** [comment]
So it looks like for this warning it was done on purpose: https://github.com/llvm/llvm-project/blob/main/clang/lib/Sema/SemaChecking.cpp#L14674

The commit that added this was 3bb8b56a5d90c535e7dc466f924d00ce798c28e0 and bug report on this: https://github.com/llvm/llvm-project/issues/16446

says that it was added to follow gcc's `-Waddress` diagnostic which also does not warn on the macro case either. 

I need to dig some more to see if this is consistent with other diagnostics.

**Comment by shafik** [comment]
If I look in `SemaExpr.cpp` we do this in a lot of diagnostics there. I also see this sprinkled around some other diagnostics. 

So I think this feels like expected behavior but not 💯 if we have a policy on where this makes sense or not. I will leave this open so if someone has deeper insights on where this does or does not make sense can chime in.
